### DISCUSSION TITLE
Ramaining

### DISCUSSION THREAD

**Discussion by Rill189** [nathanmarlor/foxess_modbus#711] [Support] [answered]
sensor.h1_fox_bms_kwh_remaining has suddenly started to show a fixed value which I can only assume is meant to be the batteries capacity rather than the remaining capacity left at it current state of charge.

The firmware on my system (Batteries etc...) was updated. Raised a call with Fox but they insist that it must be the integration as it is calculated rather than supplied from the BMS.  Is this true?

![image](https://github.com/user-attachments/assets/19a12ca4-b94f-4374-97bb-b6f44da1269c)

New Firmware versions
![image](https://github.com/user-attachments/assets/36026611-3ebf-4090-baec-5a1826bc6d14)

Thanks for any assistance.

**Comment by FozzieUK** [comment, marked as answer]
Yes it is true, the sensor no longer reports as it did, it now shows the maximum remaining kWh  of the battery pack - this post refers https://github.com/nathanmarlor/foxess_modbus/issues/651

**Reply by Rill189** [reply]
Smashing thanks for the quick update.
Is there anyway to calculate the remaining kWh capacity as it was?

Thanks

**Reply by FozzieUK** [reply]
If you scroll down that link and look at the last post in that referred thread you will see a code snippet (I think written by Tony M) which produces a new sensor based on the Fox reported sensor * battery_soc which will give you what you had before - effectively kw remaining in the battery as before.
